### PR TITLE
Added check for flw state transition

### PR DESCRIPTION
**Pull request by koshalt**
Fixes the symptoms reported in NMS-246, although the expected workflow in the bug is incorrect.

### CODE REVIEW

File: api/src/main/java/org/motechproject/nms/api/web/CallDetailsController.java [hunk 1]
```diff
@@ -106,13 +106,23 @@ public void saveCallDetails(@PathVariable String serviceName,
 
         createCallDetailRecord(flw, callDetailRecordRequest, service);
 
-        // if this is the FLW's first time calling the service, set her status to ACTIVE
-        if (flw.getStatus() == FrontLineWorkerStatus.INACTIVE) {
+        // if this is the FLW's first time calling the service, set her status to ACTIVE based on NMS.GEN.FLW.003
+        if (flw.getStatus() == FrontLineWorkerStatus.INACTIVE &&
+                validateFlwNameAndNumber(flw) &&
+                validateFlwLocation(flw)) {
```
Comment (larubbio) [inline api/src/main/java/org/motechproject/nms/api/web/CallDetailsController.java]: According to the spec we also need valid designation.  GET ON IT!

Comment (koshalt) [inline api/src/main/java/org/motechproject/nms/api/web/CallDetailsController.java]: oo

**Comment by larubbio** [comment]
lgtm.  Waiting for build

**Comment by larubbio** [comment]
ok to test

**Comment by larubbio** [comment]
ok to test

**Comment by larubbio** [comment]
ok to test

**Pull request merged**